### PR TITLE
PR for discussion: Verbose logging for token debugging

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

Relevant threads
https://community.mattermost.com/core/pl/prbb563aqbrxxjz1qsihz4i44y
https://community.mattermost.com/core/pl/5khxy7mqxi8pijprf7rid8nyuc

### CODE REVIEW

File: server/api.go [hunk 1]
```diff
@@ -34,6 +34,29 @@ const (
...
 
+	if r.URL.Path == "/set-expiry" {
+		expiry := r.URL.Query().Get("expiry")
+		userId := r.Header.Get("Mattermost-User-Id")
+		token, err := p.getGitlabUserTokenByMattermostID(userId)
+		oldExpiry := token.Expiry
+		if err != nil {
+			http.Error(w, err.Error(), 500)
+			return
+		}
```
Comment (mickmister) [inline server/api.go:45]: We can use this to set the expiry of the user's token. Run this in the browser dev console to set the expiry to 1 minute ot 0 minutes:

```js
// set to 1 minute
fetch('/plugins/com.github.manland.mattermost-plugin-gitlab/set-expiry?expiry=1')
```

```js
// set to 0 minutes
fetch('/plugins/com.github.manland.mattermost-plugin-gitlab/set-expiry?expiry=0');


```

File: server/command_test.go [hunk 2]
```diff
@@ -76,6 +76,7 @@ func TestSubscribeCommand(t *testing.T) {
 	for _, test := range subscribeCommandTests {
 		t.Run(test.testName, func(t *testing.T) {
 			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
```
Comment (hanzei) [inline server/command_test.go:79]: Woops

**Review by hanzei — CHANGES_REQUESTED** [review]
I see the value in logging the tokens. What is the deal with all of the `calling XYZ` calls? At that point, you might as well add a stacktrace.

File: server/plugin.go [hunk 3]
```diff
@@ -821,56 +887,129 @@ func (p *Plugin) handleRevokedToken(info *gitlab.UserInfo) {
...
+	log.With(logger.LogContext{"newToken": gitlab.MakeSanitizedTokenLogContext(newToken)})
+
 	if err != nil {
+		log.WithError(err).Debugf("error calling p.refreshToken")
 		return nil, err
 	}
 
+	log.Debugf("called p.refreshToken")
+
+	log.Debugf("end of getOrRefreshTokenWithMutex")
```
Comment (raghavaggarwal2308) [inline server/plugin.go:970]: I think we can keep only one log statement from these two

File: server/gitlab/api.go [hunk 4]
```diff
@@ -289,35 +290,100 @@ func (g *gitlab) GetProject(ctx context.Context, user *UserInfo, token *oauth2.T
...
+		"expiry":         token.Expiry,
+		"expiry_minutes": time.Until(token.Expiry) / time.Minute,
+	}
+}
+
+func (g *gitlab) GetLHSData(ctx context.Context, user *UserInfo, token *oauth2.Token, log logger.Logger) (*LHSContent, error) {
+	log = log.With(logger.LogContext{
+		"func":           "gitlab.getLHSData",
+		"original_token": MakeSanitizedTokenLogContext(token),
+		"token":          MakeSanitizedTokenLogContext(token),
```
Comment (raghavaggarwal2308) [inline server/gitlab/api.go:316]: We can log only one of original token and token

File: server/plugin.go (hunk 3 above)
Comment (raghavaggarwal2308) [reply server/plugin.go:970]: Similar pattern is repeated at many places

**Pull request closed without merging**